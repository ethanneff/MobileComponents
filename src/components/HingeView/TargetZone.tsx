import React, { memo } from "react";
import { Path } from "react-native-svg";
import Theme from "../../commons/Theme";
import { Point2 } from "../../commons/Utils/Maths/geometry/Point";
import { describeArc } from "../../commons/Utils/SvgUtils";

const styles = {
  targetZone: {
    color: Theme.color.neutral50,
    inTarget: {
      color: Theme.color.success
    }
  }
};

export interface TargetZoneProps {
  centerArc: Point2;
  radius: number;
  startAngle: number;
  endAngle: number;
}

interface OwnProps {
  isInTargetZone: boolean;
}

type Props = TargetZoneProps & OwnProps;

export default memo(function TargetZone(props: Props) {
  const { centerArc, radius, startAngle, endAngle, isInTargetZone } = props;
  const targetPath = describeArc(
    centerArc.x,
    centerArc.y,
    radius,
    startAngle,
    endAngle
  );

  const fillColor = isInTargetZone
    ? styles.targetZone.inTarget.color
    : styles.targetZone.color;

  return <Path id="targetZone" fill={fillColor} d={targetPath} />;
});
