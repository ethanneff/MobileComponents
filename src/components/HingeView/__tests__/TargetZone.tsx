import React from "react";
import Victor from "victor";
import { testRender } from "../../../commons/Utils/TestMocks/test-render";
import TargetZone from "../TargetZone";

describe("TargetZone Component", () => {
  it("renders correctly", () => {
    const { tree } = testRender(
      <TargetZone
        centerArc={new Victor(10, 20)}
        radius={10}
        startAngle={90}
        endAngle={120}
        isInTargetZone={true}
      />
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it("renders correctly out of targetZone", () => {
    const { tree } = testRender(
      <TargetZone
        centerArc={new Victor(10, 20)}
        radius={10}
        startAngle={90}
        endAngle={120}
        isInTargetZone={false}
      />
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
