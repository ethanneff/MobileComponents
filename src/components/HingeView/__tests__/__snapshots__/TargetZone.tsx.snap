// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TargetZone Component renders correctly 1`] = `
<RNSVGPath
  d="M 10 20 L 5 28.7 M 5 28.7 A 10 10 0 0 0 10 30 L 10 20"
  fill={4291357333}
  fillOpacity={1}
  fillRule={1}
  matrix={
    Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ]
  }
  name="targetZone"
  opacity={1}
  originX={0}
  originY={0}
  propList={
    Array [
      "fill",
    ]
  }
  rotation={0}
  scaleX={1}
  scaleY={1}
  skewX={0}
  skewY={0}
  stroke={null}
  strokeDasharray={null}
  strokeDashoffset={null}
  strokeLinecap={0}
  strokeLinejoin={0}
  strokeMiterlimit={4}
  strokeOpacity={1}
  strokeWidth={1}
  vectorEffect={0}
  x={0}
  y={0}
/>
`;

exports[`TargetZone Component renders correctly out of targetZone 1`] = `
<RNSVGPath
  d="M 10 20 L 5 28.7 M 5 28.7 A 10 10 0 0 0 10 30 L 10 20"
  fill={4293783794}
  fillOpacity={1}
  fillRule={1}
  matrix={
    Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ]
  }
  name="targetZone"
  opacity={1}
  originX={0}
  originY={0}
  propList={
    Array [
      "fill",
    ]
  }
  rotation={0}
  scaleX={1}
  scaleY={1}
  skewX={0}
  skewY={0}
  stroke={null}
  strokeDasharray={null}
  strokeDashoffset={null}
  strokeLinecap={0}
  strokeLinejoin={0}
  strokeMiterlimit={4}
  strokeOpacity={1}
  strokeWidth={1}
  vectorEffect={0}
  x={0}
  y={0}
/>
`;
